### PR TITLE
Support for setting WinRM Listeners

### PR DESCRIPTION
**Pull request by peterbeams**
Adding two new util methods to configure WinRM Listeners ConfigureWinRMOverHttp and ConfigureWinRMOverHttps(certificateThumbprint).

### CODE REVIEW

**Comment by ahmetalpbalkan** [comment]
@peterbeams Thanks for your pull request. Do you mind if we just wait @paulmey come from vacation on Monday and let him take a look and merge. This generally LGTM.

**Comment by peterbeams** [comment]
No problem to wait at all.

File: management/vmutils/vmutils.go [hunk 1]
```diff
@@ -99,3 +99,48 @@ func ConfigureWindowsToJoinDomain(role *vm.Role, username, password, domainToJoi
...
+			if existingListener.Protocol == protocol {
+				currentListeners[i] = listener
+				return nil
+			}
+		}
+
+		// otherwise append to list of listeners
+		newListeners := append(currentListeners, listener)
+		winconfig.WinRMListeners = &newListeners
+	}
```
Comment (paulmey) [inline management/vmutils/vmutils.go]: Shouldn't there be an error if this is not a windows config? Silent fail is maybe not the way to go... ;-)

**Comment by paulmey** [comment]
Sorry I missed this. I'm already in need of another vacation, I guess...
@peterbeams, can you return and error when no Windows configuration set is found? Also, please squash all commits into one commit.

Thanks, Paul

**Comment by paulmey** [comment]
@peterbeams are you still interested in moving this PR forward? :-)
Thanks, Paul

**Comment by ahmetalpbalkan** [comment]
@paulmey maybe merge and fix later on?

**Comment by paulmey** [comment]
SGTM

**Pull request merged**